cmd/multiFaSequenceSwap/multiFaSequenceSwap.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/vertgenlab/gonomics/bed"
+	"github.com/vertgenlab/gonomics/fasta"
+	"log"
+)
+
+// multiFaSubsequenceSwap swaps a series of specified regions from a bed file between two sequences in a MultiFa object
+// The foregroundName is the name of the sequence containing the regions in the bed elements;  backgroundName is the name of the sequence including the regions outside the bed elements
+// input of this function includes the multiFa file name, the index of the background and foreground sequences from the multiFa, and a bed object with the relevant
+// start and stop indices of the swap regions
+// it will return a file containing a Fasta slice (which includes several sequences), with one more sequence that the sequences in the original
+// multiFa; this extra sequence is the background sequence with relevant foreground spots swapped as dictated by the bed file.
+func multiFaSubsequenceSwap(inFile string, swapRegionsFile string, backgroundName, foregroundName string, outFile string, outSeqName string) {
+	var currStart, currEnd, currPos int //these variables will be used to swap regions later in the code
+	// Load the original sequences from the multiFa file.
+	originalSeqs := fasta.Read(inFile)
+	swapRegions := bed.Read(swapRegionsFile)
+	background := getFaIndex(originalSeqs, backgroundName)
+	foreground := getFaIndex(originalSeqs, foregroundName)
+	// Ensure the specified background and foreground sequence indices are within the valid range.
+	if background < 0 || background >= len(originalSeqs) || foreground < 0 || foreground >= len(originalSeqs) {
+		log.Fatalf("Error: Invalid sequence index.\n")
+
+	}
+	answerSeq := fasta.Copy(originalSeqs[background]) //copy of the background sequence is made
+	for _, swapRegion := range swapRegions {          //loop over every single swap region in the bed
+		currStart = swapRegion.ChromStart
+		currEnd = swapRegion.ChromEnd
+
+		// Validate the swap region.
+		if currStart < 0 || currEnd > len(answerSeq.Seq) || currStart >= currEnd {
+			log.Fatalf("Error: Invalid swap region. \n")
+
+		}
+
+		// Swap the corresponding portion of answerSeq with the foreground sequence.
+		for currPos = currStart; currPos < currEnd; currPos++ {
```

Comment:
I think this math does not account for gaps in the reference genome, which would be needed for the bed coordinates to always correspond to the same base in the reference genome.  It looks like right now, the bed region is being used as alignment position instead of coordinates in the reference genome.